libkbfs/mdserver_remote.go

```diff
@@ -387,6 +388,20 @@ func (md *MDServerRemote) ShouldRetryOnConnect(err error) bool {
 	return !inputCanceled
 }
 
+// CheckReachability implements the MDServer interface.
+func (md *MDServerRemote) CheckReachability(ctx context.Context) {
+	conn, err := net.DialTimeout("tcp", md.mdSrvAddr, MdServerPingTimeout)
+	if conn != nil {
+		conn.Close()
+		return
+	}
+	if err != nil {
```

Comment:
Can we re-order these cases so the err check is first?  Seems more natural.  Also, if `err == nil`, will `conn` ever be `nil`?